### PR TITLE
[kfr] Add new port

### PR DESCRIPTION
**Pull request by myd7349**

### CODE REVIEW

File: ports/kfr/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
+include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/kfr/portfile.cmake]: include(vcpkg_common_functions) is deprecated. Please remove it.

File: ports/kfr/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        -DKFR_EXTENDED_TESTS=OFF
+        -DSKIP_TESTS=ON
+)
+
+vcpkg_install_cmake()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+
+# Handle copyright
+configure_file(${SOURCE_PATH}/LICENSE.txt ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)
```
Comment (LilyWangL) [inline ports/kfr/portfile.cmake]: Can you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by LilyWangL** [comment]
Pinging @myd7349 for response. Is work still being done for this PR?

**Comment by LilyWangL** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Comment by myd7349** [comment]
Hi! @LilyWangL  Sorry for the long delay.
I have finished all my other PRs and I think it is time to pick this one up. But I am not able to reopen it.
Can you help me reopen it? Thanks.

**Comment by LilyWangL** [comment]
> Hi! @LilyWangL Sorry for the long delay.
> I have finished all my other PRs and I think it is time to pick this one up. But I am not able to reopen it.
> Can you help me reopen it? Thanks.

Done.

**Comment by LilyWangL** [comment]
Hi, @myd7349 Now `kfr` build failed on Linux with the following error:
```
/mnt/vcpkg-ci/buildtrees/kfr/src/70ebd9a223-37f5ac66e1/include/kfr/io/impl/../../base/../simd/impl/../../base/../simd/impl/backend_generic.hpp:653:72: error: macro "_mm256_shuffle_ps" passed 7 arguments, but takes just 3
     return _mm256_shuffle_ps(t1, t2, shuffle_mask<8, 2, 3, 2, 3>::value);
                                                                        ^
/mnt/vcpkg-ci/buildtrees/kfr/src/70ebd9a223-37f5ac66e1/include/kfr/io/impl/../../base/../simd/impl/../../base/../simd/impl/backend_generic.hpp:661:72: error: macro "_mm256_shuffle_ps" passed 7 arguments, but takes just 3
     return _mm256_shuffle_ps(t1, t2, shuffle_mask<8, 0, 1, 0, 1>::value);
                                                                        ^
/mnt/vcpkg-ci/buildtrees/kfr/src/70ebd9a223-37f5ac66e1/include/kfr/io/impl/../../base/../simd/impl/../../base/../simd/impl/backend_generic.hpp:1655:82: error: macro "_mm256_shuffle_ps" passed 7 arguments, but takes just 3
             return _mm256_shuffle_ps(x, x, shuffle_mask<8, I0, I1, I2, I3>::value);
                                                                                  ^
/mnt/vcpkg-ci/buildtrees/kfr/src/70ebd9a223-37f5ac66e1/include/kfr/io/impl/../../base/../simd/impl/../../base/../simd/impl/backend_generic.hpp:1676:75: error: macro "_mm256_blend_ps" passed 11 arguments, but takes just 3
                                             1 - I6 / 4, 1 - I7 / 4>::value);
                                                                           ^
```

**Comment by myd7349** [comment]
> 
> 
> Hi, @myd7349 Now `kfr` build failed on Linux with the following error:
> 
> ```
> /mnt/vcpkg-ci/buildtrees/kfr/src/70ebd9a223-37f5ac66e1/include/kfr/io/impl/../../base/../simd/impl/../../base/../simd/impl/backend_generic.hpp:653:72: error: macro "_mm256_shuffle_ps" passed 7 arguments, but takes just 3
>      return _mm256_shuffle_ps(t1, t2, shuffle_mask<8, 2, 3, 2, 3>::value);
>                                                                         ^
> /mnt/vcpkg-ci/buildtrees/kfr/src/70ebd9a223-37f5ac66e1/include/kfr/io/impl/../../base/../simd/impl/../../base/../simd/impl/backend_generic.hpp:661:72: error: macro "_mm256_shuffle_ps" passed 7 arguments, but takes just 3
>      return _mm256_shuffle_ps(t1, t2, shuffle_mask<8, 0, 1, 0, 1>::value);
>                                                                         ^
> /mnt/vcpkg-ci/buildtrees/kfr/src/70ebd9a223-37f5ac66e1/include/kfr/io/impl/../../base/../simd/impl/../../base/../simd/impl/backend_generic.hpp:1655:82: error: macro "_mm256_shuffle_ps" passed 7 arguments, but takes just 3
>              return _mm256_shuffle_ps(x, x, shuffle_mask<8, I0, I1, I2, I3>::value);
>                                                                                   ^
> /mnt/vcpkg-ci/buildtrees/kfr/src/70ebd9a223-37f5ac66e1/include/kfr/io/impl/../../base/../simd/impl/../../base/../simd/impl/backend_generic.hpp:1676:75: error: macro "_mm256_blend_ps" passed 11 arguments, but takes just 3
>                                              1 - I6 / 4, 1 - I7 / 4>::value);
>                                                                            ^
> ```

Hi! @LilyWangL Thanks. I will try to fix it.

**Review by LilyWangL — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR!

**Comment by soulthreads** [comment]
Here's a problem: KFR doesn't compile with MSVC compiler if you enable DFT: https://github.com/kfrlib/kfr/issues/70

Not sure if there is even a way to do this in vcpkg, but it needs to be built with clang-cl compiler on Windows.

**Pull request merged**